Signal/src/ViewControllers/Registration/CodeVerificationViewController.m

```diff
@@ -112,16 +112,25 @@ - (void)createViews
     [titleLabel autoSetDimension:ALDimensionHeight toSize:40];
     [titleLabel autoHCenterInSuperview];
 
-    UIButton *backButton = [UIButton buttonWithType:UIButtonTypeCustom];
-    [backButton
-        setTitle:NSLocalizedString(@"VERIFICATION_BACK_BUTTON", @"button text for back button on verification view")
-        forState:UIControlStateNormal];
-    [backButton setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
-    backButton.titleLabel.font = [UIFont ows_mediumFontWithSize:14.f];
-    [header addSubview:backButton];
-    [backButton autoPinLeadingToSuperviewMarginWithInset:10.f];
-    [backButton autoAlignAxis:ALAxisHorizontal toSameAxisOfView:titleLabel];
-    [backButton addTarget:self action:@selector(backButtonPressed:) forControlEvents:UIControlEventTouchUpInside];
+    // This view is used in more than one context.
+    //
+    // * Usually, it is pushed atop RegistrationViewController in which
+    //   case we want a "back" button.
+    // * It can also be used to re-register from the app's "de-registration"
```

Comment:
>  from the app's "de-registration"

Do you mean "re-registration" ?

Reply:
Sorry, these changes introduce two concepts:

* De-registration - when the app receives 403 from the service, it becomes "de-registered" and warns the user of that.
* Re-registration - the process of registering again to recover from de-registration.  Unlike registering for the first time, re-registering requires users to use the same phone number.

Reply:
Right - so if we are in the CodeVerificationViewController, aren't we in the "re-registration" flow?

Reply:
Sorry for the lack of clarity.  Users _start_ the re-registration process from two places: app settings and home view's de-registration warning UIs.  When shown from these places, we skip the "registration view" since we already know the phone number and skip to the "code verification" view.  Since there's no "registration view" in these two UI flows, there's no back button on the "code verification" view.